packages/core/src/utils/langchain/utils.ts

```diff
@@ -0,0 +1,424 @@
+import { SEMANTIC_ATTRIBUTE_SENTRY_ORIGIN } from '../../semanticAttributes';
+import type { SpanAttributeValue } from '../../types-hoist/span';
+import {
+  GEN_AI_OPERATION_NAME_ATTRIBUTE,
+  GEN_AI_REQUEST_FREQUENCY_PENALTY_ATTRIBUTE,
+  GEN_AI_REQUEST_MAX_TOKENS_ATTRIBUTE,
+  GEN_AI_REQUEST_MESSAGES_ATTRIBUTE,
+  GEN_AI_REQUEST_MODEL_ATTRIBUTE,
+  GEN_AI_REQUEST_PRESENCE_PENALTY_ATTRIBUTE,
+  GEN_AI_REQUEST_STREAM_ATTRIBUTE,
+  GEN_AI_REQUEST_TEMPERATURE_ATTRIBUTE,
+  GEN_AI_REQUEST_TOP_P_ATTRIBUTE,
+  GEN_AI_RESPONSE_FINISH_REASONS_ATTRIBUTE,
+  GEN_AI_RESPONSE_ID_ATTRIBUTE,
+  GEN_AI_RESPONSE_MODEL_ATTRIBUTE,
+  GEN_AI_RESPONSE_STOP_REASON_ATTRIBUTE,
+  GEN_AI_RESPONSE_TEXT_ATTRIBUTE,
+  GEN_AI_RESPONSE_TOOL_CALLS_ATTRIBUTE,
+  GEN_AI_SYSTEM_ATTRIBUTE,
+  GEN_AI_USAGE_CACHE_CREATION_INPUT_TOKENS_ATTRIBUTE,
+  GEN_AI_USAGE_CACHE_READ_INPUT_TOKENS_ATTRIBUTE,
+  GEN_AI_USAGE_INPUT_TOKENS_ATTRIBUTE,
+  GEN_AI_USAGE_OUTPUT_TOKENS_ATTRIBUTE,
+  GEN_AI_USAGE_TOTAL_TOKENS_ATTRIBUTE,
+} from '../ai/gen-ai-attributes';
+import { LANGCHAIN_ORIGIN, ROLE_MAP } from './constants';
+import type { LangChainLLMResult, LangChainMessage, LangChainSerialized } from './types';
+
+/**
+ * Assigns an attribute only when the value is neither `undefined` nor `null`.
+ *
+ * We keep this tiny helper because call sites are repetitive and easy to miswrite.
+ * It also preserves falsy-but-valid values like `0` and `""`.
+ */
+const setIfDefined = (target: Record<string, SpanAttributeValue>, key: string, value: unknown): void => {
+  if (value !== undefined && value !== null) target[key] = value as SpanAttributeValue;
```

Comment:
m: We typically shorten this check to just. Should also help with bundlesize a bit.

```suggestion
  if (value != null) target[key] = value as SpanAttributeValue;
```